records/labels/labels.go

```diff
@@ -0,0 +1,83 @@
+package labels
+
+import (
+	"bytes"
+	"strings"
+)
+
+// Sep is the default domain fragment separator.
+const Sep = "."
+
+// DomainFrag mangles the given name in order to produce a valid domain fragment.
+// A valid domain fragment will consist of one or more host name labels
+// concatenated by the given separator.
+func DomainFrag(name, sep string, label Func) string {
+	var labels []string
+	for _, part := range strings.Split(name, sep) {
+		if lab := label(part); lab != "" {
+			labels = append(labels, lab)
```

Comment:
Of course. But throwing around with arrays and closures when a local optimization is not more complex....